src/components/NSLookupTool/NSLookupTool.tsx

```diff
@@ -60,70 +64,65 @@ export function NSLookup() {
+        setHasSaved(true); // Mark output as saved
+        setAllowSave(false); // Disabling the Save option
     };
 
-    const onSubmit = (values: FormValues) => {
-        // Disallow saving until the tool's execution is complete
-        setAllowSave(false);
+    // Handling the submitted form
 
-        setLoading(true);
-        const args = ["-s", values.ipaddress];
-        CommandHelper.runCommandGetPidAndOutput("nslookup", args, handleProcessData, handleProcessTermination)
+    const onSubmit = (values: FormValuesType) => {
+        const ipAddress = values.ipAddress.trim(); // Removing the extra white space
+        if (!ipAddress) {
+            setOutput("Error: No IP address or hostname provided."); // Error Message
+            return;
+        }
+
+        setAllowSave(false); // Disabling the save option before start
+        setLoading(true); // Show loading indication
+
+        CommandHelper.runCommandGetPidAndOutput("nslookup", [ipAddress], handleProcessData, handleProcessTermination)
             .then(({ pid, output }) => {
-                setPid(pid);
-                setOutput(output);
+                setPid(pid); // Setting the Process ID
+                setOutput(output); // Setting the command output
             })
             .catch((error) => {
-                setLoading(false);
-                setOutput(`Error: ${error.message}`);
+                setLoading(false); //Stop Loading
+                setOutput(`Error: ${error.message}`); // Display the error message
             });
     };
 
-    /**
-     * clearOutput: A callback function to clear the current output displayed in the UI.
-     * This function resets the output state to an empty string.
-     */
+    // Clearing the output
+
     const clearOutput = useCallback(() => {
-        setOutput("");
-        setHasSaved(false);
-        setAllowSave(false);
-    }, [setOutput]); // Dependency on the setOutput function.
+        setOutput(""); // Clears output
+        setHasSaved(false); // Reset save status
+        setAllowSave(false); // Disable saving
+    }, [setOutput]);
 
     return (
         <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
             {LoadingOverlayAndCancelButton(loading, pid)}
             <Stack>
-                {UserGuide(title, description_userguide)}
+                {UserGuide(title, description_Userguide)}
                 <TextInput
-                    label={"Please enter the IP Address for nslookup"}
+                    label={"Please enter the IP Address for NSLookup"}
                     required
-                    {...form.getInputProps("ipaddress")}
+                    {...form.getInputProps("ipAddress")}
                 />
+
+                <TextInput label={"Tutorial"} {...form.getInputProps("tutorial")} />
+
                 {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
                 <Button type={"submit"}>Scan</Button>
```

Comment:
Convention is "Start <title of tool>" 
```suggestion
                <Button type={"submit"}>Start {Title}</Button>
```